test/test_optim.py

```diff
@@ -140,6 +140,63 @@ def test_set_default_dtype_works_with_foreach(self, device, dtype, optim_info):
                 torch.set_default_dtype(old_default_dtype)
 
 
+    @onlyCUDA
+    @optims([optim for optim in optim_db if "foreach" in optim.supported_impls], dtypes=[torch.float32])
+    def test_peak_memory_foreach(self, device, dtype, optim_info):
+        nparams = 10
+        optim_inputs = optim_info.optim_inputs_func()
+        optim_cls = optim_info.optim_cls
+        for optim_input in optim_inputs:
+            kwargs = deepcopy(optim_input.kwargs)
+            if kwargs.get("capturable", False) and optim_cls.__name__ == "ASGD":
```

Comment:
nit: do we want to encore skips in the test here or as a skip on the OptimInfo?

Reply:
This is a good point, I'll look into that. I see this as an unfortunate skip vs a necessary skip. Are DecorateInfos used mostly for permanent skips or temporary ones?

Reply:
I think one of the feedback we had from OpInfo is that it is confusing to have skips in different places and they should be centralized.
I think we should strive to get there for this new version as well.

Reply:
Okay, I'll put everything in DecorateInfo for now